models/groups.go

```diff
@@ -167,10 +167,13 @@ func (g *CreateUpdateGroup) ValidateCreateUpdateGroupRequest(ctx context.Context
 
 	if g.Name == nil {
 		validationErrs = append(validationErrs, NewValidationError(ctx, InvalidGroupName, MissingGroupName))
-	} else if m, _ := regexp.MatchString("(?i)^role-.*", *g.Name); m {
-		validationErrs = append(validationErrs, NewValidationError(ctx, InvalidGroupName, IncorrectPatternInGroupName))
 	} else {
-		//ensure group name in description doesn't already exist - creation only - g.GroupsList not set on updates
+		if m, _ := regexp.MatchString("(?i)^role-.*", *g.Name); m {
```

Comment:
This has changed the semantics. The following `else` code now always runs, even if the regex matches whereas before it didn't. 
Review this and fix.